### PR TITLE
Slightly improved Dockerfile

### PR DESCRIPTION
**Pull request by AsavarTzeth**
I made two minor improvements to the Dockerfile.

My reasons for doing this was to make it easier to use ngnix as a base image. In my case to make a series of packaged web-apps.
- Switched the nginx config back to default location /usr/local/nginx/conf/. Reason being it is easier to use it with volumes. /etc/nginx/ would work as well, but since it is built from source /usr/local is more appropriate.
- Added removal of source files. It may only be around 42MB but it does not belong in the production image.

### CODE REVIEW

**Comment by sprin** [comment]
The first issue was a minor annoyance to me as well, but I worked around it with two volume mounts: one for /etc/nginx.conf, and another for /etc/nginx/ (where conf.d lives):

```
docker run -d \
    -v /var/docker_volumes/nginx/nginx.conf:/etc/nginx.conf \
    -v /var/docker_volumes/nginx/nginx:/etc/nginx \
    nginx
```

Regarding the second point, it would be great to shed 42mb from this image.

**Comment by AsavarTzeth** [comment]
I just realized the apt cache should always be emptied as well, using "apt-get clean".

Do I need to make a new pull request or could a maintainer just add it bellow my rm line?

File: Dockerfile [hunk 1]
```diff
@@ -48,7 +48,10 @@ RUN { \
 		echo; \
 		echo '# stay in the foreground so Docker has a process to track'; \
 		echo 'daemon off;'; \
-	} >> /etc/nginx.conf
+	} >> /usr/local/etc/nginx.conf
+
+# Build cleanup
+RUN rm -R /usr/src/* && apt-get clean
```
Comment (tianon) [inline Dockerfile:54]: As a RUN line by itself, this won't actually have any benefit at all thanks to the way Docker handles layering, but besides that, see also https://github.com/dotcloud/docker/blob/master/contrib/mkimage/debootstrap#L64-L87 for why the `apt-get clean` shouldn't be necessary either.

**Comment by tianon** [comment]
So just to make sure I understand, the benefit here is that you can easily bind-mount over the entire configuration with a single bind mount instead of using two?

**Comment by AsavarTzeth** [comment]
Thanks for explaining about layers. You made me realize that does indeed serve no purpose. To be even close to doing any good it would need to clean the files at the same layer that you add them or before the build, am I right? The more I think about it the more I see the impossibilities
.
Regarding the bind mount you are correct. But mainly my issue was that I
found it unnecessarily confusing.

In the end my goal is only share nginx.conf, but not all of /etc.
Den 21 jul 2014 17:35 skrev "Tianon Gravi" notifications@github.com:

**Comment by sprin** [comment]
Having to configure two mounts is not a problem here, I think it's just somewhat unexpected in this case.

What do you think of adding a README that clarifies this, and maybe could grow to include any other Docker-specific configuration patterns that might be helpful? For example, what are the tradeoffs of having the external config files live on the host, versus in a data-only container?

Regarding being unable to remove the source from the image, I guess that's a current limitation of Dockerfiles, in that you cannot ADD content without committing it! Even if you would like to remove it later. Perhaps a solution is for Dockerfiles to allow users to create explicit transaction blocks, or manually commit of sets of instructions.

Interestingly, if the content were being fetched from a remote source, instead of ADDed, you could fetch, build, and clean up in a single commit with a RUN instruction referencing an external script, or other CM tool.

Thanks for this image!

**Comment by AsavarTzeth** [comment]
I like your way of thinking. The real goal is to make it easier to use the
base image.

I find your last idea interesting. The only problem there is that you then
lose the build, debug, run process that makes working with Docker great.

I will close this now. I am satisfied, now knowing more about docker
limitations.
Den 21 jul 2014 22:16 skrev "sprin" notifications@github.com:

**Pull request closed without merging**